From 3b014db53ebb0fe9057f07b137043ad3c8309fb6 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Fri, 23 Jun 2017 16:05:53 +0300
Subject: [PATCH 2003/5242] MLK-15151: dma: imx-sdma: Fix keepings clks
 enabled on sdma_resume error

commit  70b1c408f7e5ccdb1d3d832774f837731d76472d from
https://source.codeaurora.org/external/imx/linux-imx.git

This is obviously a bug but I don't know of a scenario where those errors
might happen.

Fixes: 7e84737c9c24 ("MLK-11385 dma: imx-sdma: enable clock before context restored")
Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/imx-sdma.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index 903a197..5790c75 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -2479,19 +2479,21 @@ static int sdma_resume(struct device *dev)
 	ret = sdma_get_firmware(sdma, sdma->fw_name);
 	if (ret) {
 		dev_warn(&pdev->dev, "failed to get firware\n");
-		return ret;
+		goto out;
 	}
 
 	ret = sdma_save_restore_context(sdma, false);
 	if (ret) {
 		dev_err(sdma->dev, "restore context error!\n");
-		return ret;
+		goto out;
 	}
 
+	ret = 0;
+out:
 	clk_disable(sdma->clk_ipg);
 	clk_disable(sdma->clk_ahb);
 
-	return 0;
+	return ret;
 }
 #endif
 
-- 
1.7.9.5

